### PR TITLE
meson: let's make unified cgroupsv2 the default

### PR DESCRIPTION
**Pull request by poettering**
Prompted by @evverx' comments:

https://github.com/systemd/systemd/pull/10161#pullrequestreview-158327715

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -2,6 +2,16 @@ systemd System and Service Manager
 
 CHANGES WITH 240 in spe:
 
+        * systemd now defaults to the "unified" cgroup hierarchy setup during
+          build-time, i.e. -Ddefault-hierarchy=unfied is now the build-time
```
Comment (davide125) [inline NEWS]: You have a typo here, s/unfied/unified/

**Comment by evverx** [comment]
I didn't see this coming :-)

`TEST-13-NSPAWN-SMOKE` seems to have started failing. It should probably be fixed first before the PR is merged.

**Comment by evverx** [comment]
As far as I can tell, when `SYSTEMD_NSPAWN_USE_CGNS=no` is passed to `nspawn`, it expects `systemd` (or `machined`) to create a scope in `/sys/fs/cgroup/systemd`, which apparently doesn't happen when the unified hierarchy is used by default. So either `systemd` should maintain the legacy hierarchy (which is effectively what `hybrid` is for :-)) or `nspawn` should do it itself (in which case the point of the dbus API isn't completely clear).

**Comment by evverx** [comment]
Probably the dbus api can be extended, but I'm not sure what to do with `--keep-unit --register=no`. It will stop working if `nspawn` doesn't start maintaining the legacy hierarchy.

**Comment by poettering** [comment]
> As far as I can tell, when `SYSTEMD_NSPAWN_USE_CGNS=no` is passed to `nspawn`, it expects `systemd` (or `machined`) to create a scope in `/sys/fs/cgroup/systemd`, which apparently doesn't happen when the unified hierarchy is used by default. So either `systemd` should maintain the legacy hierarchy (which is effectively what `hybrid` is for :-)) or `nspawn` should do it itself (in which case the point of the dbus API isn't completely clear).

This has been fixed since, in #9658.

**Comment by poettering** [comment]
So, I force pushed another, rebased version, to get this restarted. This time CI seems to have passed (modulo the arm one, which appears unrelated). So I guess this can go in now.

File: NEWS [hunk 2]
```diff
@@ -27,6 +27,16 @@ CHANGES WITH 243 in spe:
         * Memory protection directives can now take a value of zero, allowing
           explicit opting out of a default value propagated by an ancestor.
 
+        * systemd now defaults to the "unified" cgroup hierarchy setup during
+          build-time, i.e. -Ddefault-hierarchy=unified is now the build-time
+          default. Previously, -Ddefault-hierarchy=hybrid was the default. This
+          change reflects the fact that cgroupsv2 has matured substantially in
+          both systemd and in the kernel, and is clearly the way
+          forward. Downstream production distributions might want to continue
+          to use -Ddefault-hierarchy=legacy for their builds as unfortunately
```
Comment (keszybz) [inline NEWS]: Hmm, why not =hybrid here?

Comment (poettering) [reply NEWS]: good point, will fix

**Comment by poettering** [comment]
updated, ptal

**Comment by keszybz** [comment]
Let's do this!

**Pull request merged**